src/bundle.c

```diff
@@ -29,15 +30,25 @@ static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GEr
 		goto out;
 	}
 
-	sproc = r_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_SILENCE,
-			&ierror, "mksquashfs",
-			contentdir,
-			bundlename,
-			"-all-root",
-			"-noappend",
-			"-no-progress",
-			"-no-xattrs",
-			NULL);
+	g_ptr_array_add(args, g_strdup("mksquashfs"));
+	g_ptr_array_add(args, g_strdup(contentdir));
+	g_ptr_array_add(args, g_strdup(bundlename));
+	g_ptr_array_add(args, g_strdup("-all-root"));
+	g_ptr_array_add(args, g_strdup("-noappend"));
+	g_ptr_array_add(args, g_strdup("-no-progress"));
+	g_ptr_array_add(args, g_strdup("-no-xattrs"));
+
+	if (r_context()->squashfs_args != NULL) {
```

Comment:
I first head to look up if the options args string is actually NULL-terminated, as I could not find any explicit mention, but this seems to be the case. For this, the rest of the RAUC code uses a more condensed way of iterating that I would propose here, too. And the if could be skipped, too.

```suggestion
      for (gchar **squashfs_args = r_context()->squashfs_args; *squashfs_args != NULL; squashfs_args++) {
              g_ptr_array_add(args, g_strdup(*squashfs_args));

```

Reply:
Given the change to g_shell_parse_argv I couldn't see an easy way to get rid of this if statement after all, as it fails if the string passed is NULL.